VCFv4.5.draft.tex

```diff
@@ -1702,6 +1741,26 @@ \subsection{Representing unspecified alleles and REF-only blocks (gVCF)}
 \normalsize
 
 
+\subsubsection{Multi-sample REF-only blocks}
+When handling VCFs with multiple samples, the length of the $<$*$>$ reference blocks can differ.
+To account for this, a sample-specific END can be specified via the FORMAT END field.
+If any FORMAT END value exists, the INFO END must be present and equal the largest FORMAT END value.
```

Comment:
After reviewing everything a second time, may I propose this instead?

> If the VCF contains exactly one sample column, the INFO END field must be present and equal to the FORMAT END value at every locus. Otherwise (zero or two or more samples), the INFO END must be missing.

The INFO END, as a single value, will always be misinterpreted by tools written for GVCF. They will either treat some reference genotypes as missing (if INFO END is the min over all samples) or treat some missing or variant genotypes as reference genotypes (if INFO END is the max over all samples). Given this, I prefer that INFO END is required to be missing unless there is exactly one sample.

Reply:
I believe the purpose of INFO END is for indexing.  VCF variants aren't point source - they can have size.  In that case for range queries to work you need a spatial index (an R-tree in the case of BAI and CSI) which needs to track start and end coordinates.  Hence setting it to the largest value guarantees that a genomic location query will always find overlapping records even if it's just in one sample.

Tools should not be naively assuming END has a meaning per sample, hence the addition of FORMAT END.